src/napari/_qt/qt_main_window.py

```diff
@@ -1239,10 +1238,38 @@ def add_dock_widget(
             layers_events.reordered.connect(widget.reset_choices)
 
         # Add dock widget to dictionary
-        self._dock_widgets[dock_widget.name] = dock_widget
+        self._wrapped_dock_widgets[dock_widget.name] = dock_widget
 
         return dock_widget
 
+    @property
+    def _dock_widgets(self) -> MutableMapping[str, QtViewerDockWidget]:
+        """Access `_wrapped_dock_widgets` with warning.
+
+        Before napari 0.6.2, ``_wrapped_dock_widgets`` was just
+        ``_dock_widgets``. Even though it was private, many
+        resources pointed to its use, as there was no public alternative.
+        Now that `dock_widgets` is provided, we want to make sure
+        that people stop using the private `_dock_widgets`.
+        """
+        # As many plugins uses `_dock_widget` to access one widget from the
+        # other widget we should keep this name for a longer period
+        warnings.warn(
+            'The `_dock_widgets` property is private and should not be used in any plugin code. '
+            'Please use the `dock_widgets` property instead.',
```

Comment:
Did we want schedule removal (for example for 0.8.0, or 0.9.0)?

Reply:
I don't have a strong preference.